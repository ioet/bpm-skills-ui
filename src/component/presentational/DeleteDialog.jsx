import React from 'react';
import Dialog from '@material-ui/core/Dialog/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions/DialogActions';
import Button from '@material-ui/core/Button/Button';
import Slide from '@material-ui/core/Slide/Slide';
import PropTypes from 'prop-types';
import { DeleteDialogConst } from '../../constants';
import { getCategoryToBeCreated } from '../utils/Utils';

function Transition(props) {
  return <Slide direction="up" {...props} />;
}

const DeleteDialog = (props) => {
  const {
    open, handleClose, categoryIds, categoryName,
  } = props;

  return (
    <Dialog
      open={open}
      TransitionComponent={Transition}
      keepMounted
      onClose={(e) => {
        e.preventDefault();
        handleClose();
      }}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">
        {DeleteDialogConst.TITLE}
      </DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {DeleteDialogConst.CONTENT_TEXT_1}
          <b>
            {categoryName}
          </b>
          {DeleteDialogConst.CONTENT_TEXT_2}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button
          onClick={(e) => {
            e.preventDefault();
            handleClose();
          }}
          color="primary"
        >
          {DeleteDialogConst.DISAGREE}
        </Button>
        <Button
          onClick={(e) => {
            e.preventDefault();
            handleClose(categoryIds);
          }}
          color="primary"
        >
          {DeleteDialogConst.AGREE}
        </Button>
      </DialogActions>
    </Dialog>
  );
};

DeleteDialog.defaultProps = {
  categoryIds: [getCategoryToBeCreated().id],
  username: [getCategoryToBeCreated().name],
};

DeleteDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  categoryIds: PropTypes.arrayOf(PropTypes.string),
  categoryName: PropTypes.string,
};

export default DeleteDialog;
