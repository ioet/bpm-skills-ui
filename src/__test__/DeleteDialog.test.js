import React from 'react';
import DeleteDialog from '../component/presentational/DeleteDialog';

describe('<DeleteDialog />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<DeleteDialog />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
