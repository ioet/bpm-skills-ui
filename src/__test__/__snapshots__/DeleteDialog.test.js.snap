// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteDialog /> Component Renders correctly 1`] = `
<WithStyles(Dialog)
  TransitionComponent={[Function]}
  aria-describedby="alert-dialog-slide-description"
  aria-labelledby="alert-dialog-slide-title"
  keepMounted={true}
  onClose={[Function]}
>
  <WithStyles(DialogTitle)
    id="alert-dialog-slide-title"
  >
    Delete category
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)
      id="alert-dialog-slide-description"
    >
      Do you really want to delete 
      <b />
       permanently?
    </WithStyles(DialogContentText)>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      No
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Yes
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
